yaksh/views.py

```diff
@@ -730,16 +730,17 @@ def monitor(request, questionpaper_id=None):
     return my_render_to_response('yaksh/monitor.html', context,
                                  context_instance=ci)
 
-
-def get_user_data(username, questionpaper_id=None):
+def get_user_data(user_id, questionpaper_id=None, attempt_number = None):
     """For a given username, this returns a dictionary of important data
     related to the user including all the user's answers submitted.
     """
-    user = User.objects.get(username=username)
+    user = User.objects.get(id=user_id)
     papers = AnswerPaper.objects.filter(user=user)
-    if questionpaper_id is not None:
-        papers = papers.filter(question_paper_id=questionpaper_id).order_by(
+    if questionpaper_id and attempt_number is not None:
+        papers = papers.filter(question_paper_id=questionpaper_id, attempt_number = attempt_number).order_by(
```

Comment:
why do we need order_by here as we are passing attempt_number and questionpaper_id?
I think the query will return only one paper.